src/analysis/inner_product_space/area_form.lean

```diff
@@ -0,0 +1,440 @@
+end
+
+/-- An isometric automorphism of an oriented real inner product space of dimension 2 (usual notation
+`J`). This automorphism squares to -1.  We will define rotations in such a way that this
+automorphism is equal to rotation by 90 degrees. -/
+def right_angle_rotation : E ≃ₗᵢ[ℝ] E :=
+linear_isometry_equiv.of_linear_isometry
+  o.right_angle_rotation_aux₂
+  (-o.right_angle_rotation_aux₁)
+  (by ext; simp [right_angle_rotation_aux₂])
+  (by ext; simp [right_angle_rotation_aux₂])
+
+local notation `J` := o.right_angle_rotation
+
+@[simp] lemma inner_right_angle_rotation_left (x y : E) : ⟪J x, y⟫ = ω x y :=
+o.inner_right_angle_rotation_aux₁_left x y
+
+@[simp] lemma inner_right_angle_rotation_right (x y : E) : ⟪x, J y⟫ = - ω x y :=
+o.inner_right_angle_rotation_aux₁_right x y
+
+@[simp] lemma right_angle_rotation_right_angle_rotation (x : E) : J (J x) = - x :=
+o.right_angle_rotation_aux₁_right_angle_rotation_aux₁ x
+
+@[simp] lemma right_angle_rotation_symm :
+  linear_isometry_equiv.symm J = linear_isometry_equiv.trans J (linear_isometry_equiv.neg ℝ) :=
+linear_isometry_equiv.to_linear_isometry_injective rfl
+
+attribute [irreducible] right_angle_rotation
+
+@[simp] lemma inner_right_angle_rotation_self (x : E) : ⟪J x, x⟫ = 0 := by simp
+
+lemma inner_right_angle_rotation_swap (x y : E) : ⟪x, J y⟫ = - ⟪J x, y⟫ := by simp
+
+lemma inner_right_angle_rotation_swap' (x y : E) : ⟪J x, y⟫ = - ⟪x, J y⟫ :=
+by simp [o.inner_right_angle_rotation_swap x y]
+
+lemma inner_comp_right_angle_rotation (x y : E) : ⟪J x, J y⟫ = ⟪x, y⟫ :=
+linear_isometry_equiv.inner_map_map J x y
+
+@[simp] lemma area_form_right_angle_rotation_left (x y : E) : ω (J x) y = - ⟪x, y⟫ :=
+by rw [← o.inner_comp_right_angle_rotation, o.inner_right_angle_rotation_right, neg_neg]
+
+@[simp] lemma area_form_right_angle_rotation_right (x y : E) : ω x (J y) = ⟪x, y⟫ :=
+by rw [← o.inner_right_angle_rotation_left, o.inner_comp_right_angle_rotation]
+
+@[simp] lemma area_form_comp_right_angle_rotation (x y : E) : ω (J x) (J y) = ω x y :=
+by simp
+
+@[simp] lemma right_angle_rotation_trans_right_angle_rotation :
+  linear_isometry_equiv.trans J J = linear_isometry_equiv.neg ℝ :=
+by ext; simp
+
+/-- For a nonzero vector `x` in an oriented two-dimensional real inner product space `E`,
+`![x, J x]` forms an (orthogonal) basis for `E`. -/
+def basis_right_angle_rotation (x : E) (hx : x ≠ 0) : basis (fin 2) ℝ E :=
+@basis_of_linear_independent_of_card_eq_finrank _ _ _ _ _ _ _ _ ![x, J x]
+(linear_independent_of_ne_zero_of_inner_eq_zero (λ i, by { fin_cases i; simp [hx] })
+  begin
+    intros i j hij,
+    fin_cases i; fin_cases j,
+    { simpa },
+    { simp },
+    { simp },
+    { simpa }
+  end)
+(fact.out (finrank ℝ E = 2)).symm
+
+@[simp] lemma coe_basis_right_angle_rotation (x : E) (hx : x ≠ 0) :
+  ⇑(o.basis_right_angle_rotation x hx) = ![x, J x] :=
+coe_basis_of_linear_independent_of_card_eq_finrank _ _
+
+lemma inner_mul_inner_add_area_form_mul_area_form' (a x : E) :
```

Comment:
can you add a dosctring to this one? It is pretty hard to parse with all the underscores.